homeassistant/components/seventeentrack/sensor.py

```diff
@@ -111,242 +108,187 @@ async def async_setup_entry(
 ) -> None:
     """Set up a 17Track sensor entry."""
 
-    client = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator: SeventeenTrackCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    current_packages: set[Package] = set()
```

Comment:
Since the content of the set is unique, i am not sure if package works as type. If we can prove that this works, great, otherwise i think tracking numbers are the way to go

Reply:
Package holds the tracking number so it's unique. Also I need the Package itself for the various data it holds

Reply:
I know it's unique for us, but I'm talking about unique in the set. If a package is set to "Delivered", is that counted as a different object? If so, this code would not work

Reply:
Checkout https://github.com/home-assistant/core/blob/988c71ecc533238a1b4e82c2d9f8ec95dfd76fe0/homeassistant/components/opensky/coordinator.py#L85-L90

This is what I use in Opensky to track the flights.

Reply:
Done